contracts/SimpleCollections.sol

```diff
@@ -14,50 +15,61 @@ contract SimpleCollections is ITokenURIDescriptor, OwnableUpgradeable {
 	}
 
 	ISwapAndStake public swapAndStake;
-	address public gateway;
-	mapping(bytes32 => Image) public images;
-	mapping(uint256 => uint256) public stakedAmountAtMinted;
+	mapping(address => address) public gateway;
+	mapping(address => mapping(bytes32 => Image)) public propertyImages;
+	mapping(address => mapping(uint256 => uint256)) public stakedAmountAtMinted;
 
 	function initialize() external initializer {
 		__Ownable_init();
 	}
 
+	modifier onlyPropertyAuthor(address _property) {
+		address author = IProperty(_property).author();
+		require(author == _msgSender(), "illegal access");
+		_;
+	}
+
 	function image(
 		uint256 id,
 		address,
 		ISTokensManagerStruct.StakingPositions memory _positions,
 		ISTokensManagerStruct.Rewards memory,
 		bytes32 key
 	) external view returns (string memory) {
-		Image memory img = images[key];
-		uint256 stakedAtMinted = stakedAmountAtMinted[id];
+		Image memory img = propertyImages[_positions.property][key];
+		uint256 stakedAtMinted = stakedAmountAtMinted[_positions.property][id];
 		if (stakedAtMinted > _positions.amount) {
 			return "";
 		}
 		return img.src;
 	}
 
-	function setImages(Image[] memory _images, bytes32[] memory _keys)
-		external
-		onlyOwner
-	{
+	function setImages(
+		ISTokensManagerStruct.StakingPositions memory _positions,
+		Image[] memory _images,
+		bytes32[] memory _keys
+	) external onlyPropertyAuthor(_positions.property) {
 		for (uint256 i = 0; i < _images.length; i++) {
 			Image memory img = _images[i];
 			bytes32 key = _keys[i];
-			images[key] = img;
+			propertyImages[_positions.property][key] = img;
 		}
 	}
 
-	function removeImage(bytes32 _key) external onlyOwner {
-		delete images[_key];
+	function removeImage(
+		ISTokensManagerStruct.StakingPositions memory _positions,
+		bytes32 _key
+	) external onlyPropertyAuthor(_positions.property) {
+		delete propertyImages[_positions.property][_key];
 	}
 
-	function setGateway(address _contract, address _gateway)
-		external
-		onlyOwner
-	{
+	function setGateway(
+		ISTokensManagerStruct.StakingPositions memory _positions,
+		address _contract,
+		address _gateway
+	) external onlyPropertyAuthor(_positions.property) {
 		swapAndStake = ISwapAndStake(_contract);
```

Comment:
`swapAndStake` is a shared value for all users, and users should not update the value. It can be initialized with the `initialize` function. (The current contract was not designed as a shared contract)